CDCgov/prime-reportstream
# 9395: deliveries v1 api

This PR adds the V1 API for fetching delivered reports to receivers

Test Steps:
1. Make sure that your DB has some reports recorded that were sent to a receiver
2. Use the following to test the API
```sh
curl --location 'http://localhost:7071/api/v1/receivers/ignore.HL7/deliveries' \
--header 'Content-Type: application/json' \
--data '{
    "sort": {
        "direction": "ASC",
        "property": "number_of_tests"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": [
        {
            "value": "2023-05-21T00:00:00+00:00",
            "filterName": "UNTIL"
        }
    ]
}'
```
Refer to the `delivery.yml` file for the parameters that you can use with the API
3. Verify that the returned report IDs were delivered to the receiver
4. Verify that the data on provider, facility, sender and test numbers are all correct for the delivered reports (I tested this by sending data checking the results, resetting the db and sending data with different provider, facility, etc.)


## Changes
- Adds a V1 API for fetching reports delivered to receivers

## Checklist

### Testing

### Process

## Linked Issues
- #9395 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
